### PR TITLE
Fix issues detected by cfn linters

### PR DESCRIPTION
**Pull request by demartinofra**
`tox -e cfn-format-check -e cfn-lint`

Used `tox -e cfn-format` to format the files

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cloudformation/aws-parallelcluster.cfn.json [hunk 1]
```diff
@@ -209,12 +207,6 @@
         "Tenancy": {
           "default": "tenancy"
         },
-        "EphemeralKMSKeyId": {
-          "default": "ephemeral_kms_key_id"
-        },
-        "ClusterReadyScript": {
-          "default": "cluster_ready"
```
Comment (enrico-usai) [inline cloudformation/aws-parallelcluster.cfn.json]: After this patch remember to cleanup the cookbook according to this:
- [files/default/compute_ready](https://github.com/aws/aws-parallelcluster-cookbook/blob/v1.6.0/files/default/compute_ready)
- [recipes/_prep_env.rb](https://github.com/aws/aws-parallelcluster-cookbook/blob/v1.6.0/recipes/_prep_env.rb#L45)

File: cloudformation/batch-substack.cfn.json [hunk 2]
```diff
@@ -418,8 +418,6 @@
     "ManageDockerImagesCustomResource": {
       "Type": "AWS::CloudFormation::CustomResource",
       "DependsOn": [
-        "CodeBuildDockerImageBuilderProject",
```
Comment (enrico-usai) [inline cloudformation/batch-substack.cfn.json]: Why are we removing these dependencies?

File: cloudformation/batch-substack.cfn.json [hunk 3]
```diff
@@ -618,8 +615,7 @@
         },
         "Runtime": "python3.6",
         "Timeout": 60
-      },
-      "DependsOn": "SendBuildNotificationFunctionExecutionRole"
```
Comment (enrico-usai) [inline cloudformation/batch-substack.cfn.json]: same as before

File: cloudformation/batch-substack.cfn.json (hunk 2 above)
Comment (demartinofra) [reply cloudformation/batch-substack.cfn.json]: Cause when the resource is already referenced in another field it is unneeded to explicitly add it again in the depends on block.

File: cloudformation/aws-parallelcluster.cfn.json (hunk 1 above)
Comment (demartinofra) [reply cloudformation/aws-parallelcluster.cfn.json]: Very good point! Thanks

File: cli/pcluster/cfnconfig.py [hunk 4]
```diff
@@ -490,8 +490,6 @@ def __init_cluster_options():
             s3_read_resource=('S3ReadResource', None),
             s3_read_write_resource=('S3ReadWriteResource', None),
             tenancy=('Tenancy', None),
-            ephemeral_kms_key_id=('EphemeralKMSKeyId', None),
```
Comment (lukeseawalker) [inline cli/pcluster/cfnconfig.py]: I would like to have this in another PR together with the template changes

Comment (demartinofra) [reply cli/pcluster/cfnconfig.py]: Will do. I thought that splitting the changes in separate commits was good enough but probably it's worth having a separate PR for this.

**Pull request merged**